r/R/compute.R

```diff
@@ -80,6 +80,36 @@ collect_arrays_from_dots <- function(dots) {
   ChunkedArray$create(!!!arrays)
 }
 
+#' @export
+quantile.ArrowDatum <- function(x,
+                                probs = seq(0, 1, 0.25),
+                                na.rm = FALSE,
+                                interpolation = c("linear", "lower", "higher", "nearest", "midpoint"),
+                                ...) {
```

Comment:
Should I add a docs entry? It doesn't seem necessary except for the interpolation options.

Reply:
can/should you accept the `type` parameter that `quantile` takes and map to the interpolation methods where appropriate?

We don't have docs for the other compute function methods, but they also don't take additional Arrow arguments IIRC. I don't think docs would help here (if you type `?quantile` you'd have to choose which help page to load, which usually is confusing IME), I'd just make sure that we link to the appropriate C++ docs somewhere (do we need another compute-specific vignette, or save this for when dplyr supports aggregation?)

Reply:
>can/should you accept the type parameter that quantile takes and map to the interpolation methods where appropriate?
I was afraid you would ask that 😂

Reply:
Ok to defer to a followup, though if you do, it would be good in this PR if our default was the same as the R default (whichever it is) and we errored if the user specifies a non-default value for `type`

Reply:
In 188ceea3fbc037c3944ac936e9492f9f388f3c95, I added an error if the user specifies a non-default value for `type`

I read through the `quantile` docs and the associated Rob Hyndman paper, but most of the sample quantile types described there are quite different from any of the options implemented in Arrow, so that did not help much. Doing quantitative comparisons was more fruitful. Here's what I found:

- The R default `type = 7` corresponds most closely to the Arrow default `interpolation = "linear"` (which is very good)
- R's `type = 7` seems to correspond closely to Arrow's `interpolation = "lower"`
- None of the other R `type` options and Arrow `interpolation` options exhibit any close correspondence

Based on these findings, and considering how few users will likely attempt to compute quantiles with Arrow using anything but the default type, I don't think there's any immediate action to take here, and I don't really think any follow-up is needed. It seems like a long shot that we would ever implement any of these other quantile algorithms